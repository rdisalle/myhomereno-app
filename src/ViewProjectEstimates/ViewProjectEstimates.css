.ViewProjectEstimates_list {
    color: black;
    list-style-type: none;
    padding: 10px;
    margin-right: 10px;
    margin-left: 10px;
    margin-top: 0;
}

.ViewProjectEstimates_name {
    border: 5px #6B98BF solid;
    width: 335px;
    margin-right: auto;
    margin-left: auto;
    margin-bottom: 0;
}

.ProjectEstimate_add {
    color: #BF612A;
    padding-right: 10px;
    padding-left: 10px;
    font-size: 20px;
    margin-top: 25px;
}

.ViewProjectEstimates {
    margin-top: 20px;
}

.ViewProjectEstimates_estimate {
    border-bottom: 5px #587AA6 solid;
    margin-bottom: 10px;
}

.ViewProjectEstimate_viewLink {
    color: white;
    font-size: 16px;
    text-decoration: none;
    padding-right: 5px;
    padding-left: 5px;
    background-color: #F2AB6D;
}

.EstimatePage_list {
    margin-top: 10px;
    margin-bottom: 10px;
}