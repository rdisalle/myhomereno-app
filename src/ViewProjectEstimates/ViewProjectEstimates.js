import React, { Component } from 'react';
import './ViewProjectEstimates.css';
import EstimatePage from '../EstimatePage/EstimatePage';
import { Link } from 'react-router-dom';
import MyHomeRenoContext from '../MyHomeRenoContext';
import PropTypes from 'prop-types' ;

class ViewProjectEstimates extends Component {
  static defaultProps = {
    match: {
      params: {}
    }
  };

  static contextType = MyHomeRenoContext;

  onDelete = () => {
    const { projectId } = this.props.match.params;
    this.props.history.push(`/view-project-estimates/${projectId}`);
  };

  getEstimatesForProject = (estimates=[], projectId) => (
    (!projectId)
        ? estimates
        : estimates.filter(estimate => estimate.project_id === JSON.parse(projectId))
  );

  findProject = (projects=[], projectId) =>
  projects.find(project => project.id === JSON.parse(projectId));

  render() {
      const { projectId } = this.props.match.params;
      const { estimates=[] } = this.context;
      const estimatesForProject = this.getEstimatesForProject(estimates, projectId);
      const { projects } = this.context;
      const project = this.findProject(projects, projectId) || { content: '' };
    return (
      <div className="ViewProjectEstimates">
        <Link className="ProjectEstimate_add" to={`/add-estimate/${projectId}`}>
          Add Estimate
        </Link>
        <h2 className="ViewProjectEstimates_name">
          {project.name}
        </h2>
        <ul className="ViewProjectEstimates_list">  
          {estimatesForProject.map(estimate =>
          <li className="ViewProjectEstimates_estimate" key={estimate.id}>
            <Link className="ViewProjectEstimate_viewLink" to={`/view-estimate-page/${estimate.id}`}>
              View 
            </Link>
            <EstimatePage
              id={estimate.id}
              name={estimate.name}
              contractor_name={estimate.contractor_name}
              price={estimate.price}
              details={estimate.details}
              total_time={estimate.total_time}
              date_created={estimate.date_created}
            />
          </li>
          )}
        </ul>
      </div>
    );
  };
}


ViewProjectEstimates.propTypes = {
  match: PropTypes.object
};

export default ViewProjectEstimates;
