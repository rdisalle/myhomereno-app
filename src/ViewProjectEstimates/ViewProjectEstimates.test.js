import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ViewProjectEstimates from './ViewProjectEstimates';

describe(`ViewProjectEstimates component`, () => {
  const props = {
    estimates: [
      {
        "id": "cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1",
        "name": "Estimate One",
        "project_id": "b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1",
        "contractor_name": "Mike's Plumbing",
        "price": "10,000",
        "details": "Can start next week",
        "total_time": "Four Weeks"
      },
      {
        "id": "cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1",
        "name": "Estimate Two",
        "project_id": "b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1",
        "contractor_name": "Mike's Plumbing",
        "price": "10,000",
        "details": "Can start next week",
        "total_time": "Four Weeks"
      },
      {
        "id": "cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1",
        "name": "Estimate Three",
        "project_id": "b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1",
        "contractor_name": "Mike's Plumbing",
        "price": "10,000",
        "details": "Can start next week",
        "total_time": "Four Weeks"
      },
      {
        "id": "cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1",
        "name": "Estimate Four",
        "project_id": "b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1",
        "contractor_name": "Mike's Plumbing",
        "price": "10,000",
        "details": "Can start next week",
        "total_time": "Four Weeks"
      },
      {
        "id": "cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1",
        "name": "Estimate Five",
        "project_id": "b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1",
        "contractor_name": "Mike's Plumbing",
        "price": "10,000",
        "details": "Can start next week",
        "total_time": "Four Weeks"
      },
    ]
  };

  it('renders a .ViewProjectEstimates by default', () => {
    const wrapper = shallow(<ViewProjectEstimates />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders a estimate in ul for each estimate in array', () => {
    const ul = shallow(<ViewProjectEstimates {...props} />)
      .find('ul');
    expect(toJson(ul)).toMatchSnapshot();
  });
});
