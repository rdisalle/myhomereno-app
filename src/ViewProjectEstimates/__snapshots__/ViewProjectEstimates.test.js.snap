// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewProjectEstimates component renders a .ViewProjectEstimates by default 1`] = `
<div
  className="ViewProjectEstimates"
>
  <h2
    className="ViewProjectEstimates_project"
  />
  <Link
    className="ProjectEstimate_add"
    to="/add-estimate/undefined"
  >
    Add Estimate
  </Link>
  <ul
    className="ViewProjectEstimates_list"
  />
</div>
`;

exports[`ViewProjectEstimates component renders a estimate in ul for each estimate in array 1`] = `
<ul
  className="ViewProjectEstimates_list"
/>
`;
